mypy/semanal.py

```diff
@@ -564,28 +564,32 @@ def check_function_signature(self, fdef: FuncItem) -> None:
 
     def visit_class_def(self, defn: ClassDef) -> None:
         self.clean_up_bases_and_infer_type_variables(defn)
-        if self.analyze_namedtuple_classdef(defn):
-            return
-        self.setup_class_def_analysis(defn)
+        is_named_tuple = self.analyze_namedtuple_classdef(defn)
 
-        self.bind_class_type_vars(defn)
+        if not is_named_tuple:
```

Comment:
Sorry for coming up with new comments, but I think that this will be cleaner if you refactor this as:
```python
if self.analyze_namedtuple_classdef(defn):
    # quick check
    self.enter_class(defn)
    defn.defs.accept(self)
    self.leave_class()
else:
    # full story
    ...
```